### ISSUE TITLE
Unified Inbox list item tap selects aleatory items 

### ISSUE THREAD

**Issue by rdoi** [k9mail/k-9#2233]
When tapping items in Unified Inbox list, an aleatory item is selected (probably mixed item indexing issue).
This works correctly in individual accound inbox list. And worked correctly in the previous releases.

### Expected behavior
In "Unified Inbox" list, when you tap in a list item/message, the item should be selected and message opened. When tap in the checkbox, the related item should be "checked".

### Actual behavior
When tapping a list item, it is selecting an aleatory item, not the tapped one. Sometimes selects the previous or the next item of the tapped one.

### Steps to reproduce
1. Open the "Unified Inbox" list (that aggregates more than one account, with mixed message dates).
2. Tap an item in the middle of the list.
3. An aleatory item is selected.

### Environment
K-9 Mail version: 5.203

Android version: 6.0.1 (Marshmallow) 
Kernel: 3.4.0-perf-gc14c2d5 / Build: 23.5.A.1.291

Account type (IMAP, POP3, WebDAV/Exchange):
POP3

**Comment by philipwhiuk** [comment]
I can't reproduce this with 2 accounts added to K-9. :/

**Comment by ozziebiker** [comment]
aléatoire? "random"

**Comment by rdoi** [comment]
Sorry, I just noticed the follow up now. 

Unfortunately I have uninstalled the app just a couple of weeks after the report and also have another device.

I am not sure if it is the same version but I just installed it again and now it is just working fine. 
It was probably an corrupted db index on cache in my old device. 

I will close this because I can not replicate it anymore.
Really sorry for bothering you with this.

PS.: Yes "aleatory" = "random" (sorry, so many languages to deal with...)
